src/frontend/views/Attester/Attester.tsx

```diff
@@ -168,14 +187,60 @@ function Connect({ setSession }: { setSession: (s: Session) => void }) {
   const [processing, setProcessing] = useState(false);
   const [error, setError] = useState<'closed' | 'rejected' | 'unknown'>();
 
-  const handleConnectClick = useCallback(
+  const extensions = getWindowExtensions();
+
+  const [extension, setExtension] = useState<InjectedWindowProvider>();
+
+  const isEmailConfirmation = useRouteMatch(paths.emailConfirmation);
+
+  useEffect(() => {
+    if (isEmailConfirmation) {
+      extensionInput.postMessage('GET_BROADCASTED_EXTENSION');
+
+      extensionOutput.onmessage = ({ data: name }) => {
+        const extension = find(extensions, { name });
+        setExtension(extension);
+      };
+    }
+
+    if (!isEmailConfirmation) {
+      if (extensions.length === 1) {
+        setExtension(extensions[0]);
+        return;
+      }
+      for (const extension of extensions) {
+        if (extension.name === 'Sporran') {
+          setExtension(extension);
+          return;
+        }
+      }
+      setExtension(extensions[0]);
+    }
+  }, [isEmailConfirmation, extensions]);
+
+  const handleInput = useCallback(
+    (event) => {
+      const match = find(extensions, { name: event.target.value });
```

Comment:
I thing a few things could be simplified by using the original `kilt` object, and not the list of its values. In that case we’d receive lowercase `sporran` as a key, and use it to get `kilt.sporran`. The human-readable extension name will only be used to be displayed to the user.

Reply:
I went with this way because the credential API explicitly includes the function `getWindowExtensions`, but yes I like your suggestion better.

Reply:
The spec is just an example :)